spring/src/main/java/org/axonframework/spring/config/annotation/SpringBeanParameterResolverFactory.java

```diff
@@ -133,19 +85,31 @@ public void setApplicationContext(ApplicationContext applicationContext) throws
         this.applicationContext = applicationContext;
     }
 
-    private static class SpringBeanParameterResolver implements ParameterResolver<Object> {
+    private static class SpringBeanMethodParameterParameterResolver implements ParameterResolver<Object> {
 
         private final AutowireCapableBeanFactory beanFactory;
-        private final String beanName;
+        private final Executable executable;
+        private final int parameterIndex;
 
-        public SpringBeanParameterResolver(AutowireCapableBeanFactory beanFactory, String beanName) {
+        public SpringBeanMethodParameterParameterResolver(AutowireCapableBeanFactory beanFactory, Executable executable, int parameterIndex) {
             this.beanFactory = beanFactory;
-            this.beanName = beanName;
+            this.executable = executable;
+            this.parameterIndex = parameterIndex;
         }
 
         @Override
         public Object resolveParameterValue(Message<?> message) {
-            return beanFactory.getBean(beanName);
+            final Autowired ann = executable.getAnnotation(Autowired.class);
+            final boolean required = ann == null || ann.required(); // Defaults to true
+            final MethodParameter methodParameter;
+
+            if (executable instanceof Method) {
+                methodParameter = new MethodParameter((Method) executable, parameterIndex);
+            } else {
+                methodParameter = new MethodParameter((Constructor<?>) executable, parameterIndex);
+            }
+
+            return beanFactory.resolveDependency(new DependencyDescriptor(methodParameter, required), null);
```

Comment:
This should resolve to the exact same instance on each invocation. I wonder if it wouldn't be best to "cache" the result in an AtomicReference. This prevents the dependency resolution to be invoked on each handler invocation.

Reply:
@abuijze this appears to not be the case.  A bean marked `        @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)` will receive a new instance each time.  There is a test for this case, `testNewInstanceIsCreatedEachTimePrototypeResourceIsInjected` that broke when I cached in the manner you've suggested.

Reply:
Fair point. Let's abandon the idea for caching :)